tools/ckb4ibc-test/src/tests/packet.rs

```diff
@@ -0,0 +1,161 @@
+        );
+
+        // 3. listen RecvPacket event on ChainB
+        info!("wait recv_packet being found on chain_b");
+        let recv_packet =
+            listen_and_wait_packet_cell(&rt, &chain_b_url, &chain_b_config, |packet| {
+                packet.is_recv_packet()
+            })?;
+        info!("🍻 successfully find recv_packet cell on chain_b");
+
+        // 4. trigger WriteAck event on ChainB
+        info!("send write_ack transaction to chain_b");
+        let acknowledgemnt = b"pong".to_vec();
+        let write_ack_tx = generate_write_ack_transaction(
+            &rt,
+            &chain_b_config,
+            &chain_b_url,
+            &chain_b_signer,
+            recv_packet,
+            acknowledgemnt,
+        )?;
+        let hash = send_transaction(&chain_b_url, write_ack_tx)?;
+        info!(
+            "🍻 successfully sent write_ack transaction to chain_b, hash = {}",
+            hex::encode(hash)
+        );
+
+        // 5. listen AckPacket event on ChainA
+        info!("wait ack_packet being found on chain_a");
+        let ack_packet =
+            listen_and_wait_packet_cell(&rt, &chain_a_url, &chain_a_config, |packet| {
+                packet.is_ack_packet()
+            })?;
+        info!("🍻 successfully find ack_packet cell on chain_a");
+
+        // 6. comsune AckPacket cell on ChainA
+        info!("send ack_packet consume transaction to chain_a");
+        let consume_ack_packet_tx = generate_consume_ack_packet_transaction(
+            &rt,
+            &chain_a_config,
+            &chain_a_url,
+            &chain_a_signer,
+            ack_packet,
+        )?;
+        let hash = send_transaction(&chain_a_url, consume_ack_packet_tx)?;
+        info!(
+            "🍻 successfully consumed ack_packet on chain_a, hash = {}",
+            hex::encode(hash)
+        );
+
+        Ok(())
+    }
+}
+
+#[ignore]
+#[test]
+fn test_send_packet() {
+    let rt = Runtime::new().unwrap();
+    let home = env!("HOME");
+    let config_toml = std::fs::read_to_string(format!("{home}/.hermes/config.toml")).unwrap();
+    let config: Config = toml::from_str(&config_toml).unwrap();
+
+    let chain_id_a: ChainId = "ckb4ibc-0".parse().unwrap();
+    let channel_id_a: ChannelId = "channel-0".parse().unwrap();
+
+    let (chain_a_config, chain_a_url, chain_a_signer) =
+        prepare_artificials(&config, &chain_id_a, &channel_id_a).unwrap();
+
+    let message = b"ping".to_vec();
+    let send_packet_tx = generate_send_packet_transaction(
+        &rt,
+        &chain_a_config,
+        &chain_a_url,
+        &chain_a_signer,
+        message,
+    )
+    .unwrap();
+
+    use relayer::chain::ckb;
+    let rpc_client = ckb::rpc_client::RpcClient::new(
+        &chain_a_url.parse().unwrap(),
+        &chain_a_url.parse().unwrap(),
+    );
+    rt.block_on(ckb::sighash::init_sighash_celldep(&rpc_client))
+        .unwrap();
+    let hash = send_transaction(&chain_a_url, send_packet_tx).unwrap();
+    println!(
```

Comment:
A way to implement this test on ibc-test-framework is to call the contract from counterparty chain, and check the contract's state in the deployed chain.

We should implement a contract both in solidity and in CKB to do this test.